### PR TITLE
[DB-39487, DB-39488, DB-39501] Fix undesired logging, Undesired excep…

### PR DESCRIPTION
**Pull request by dwhite-nuodb**
…tions for liveliness checks, Augment excpetion content

PINO was added to the Node.js driver for logging purposes.  It assumed the PINO property called "enabled" would be respected all the time, but it turns out to be overidden if the PINO level propery is set.  To protect from this, any place the PINO level property is set, we need to only do it if the enabled property is true, which is based on the environment variable PINO_LOG_ENABLED.

3DMessage as seeing exceptions trying to close exceptions during liveliness checks, but we don't have an exact use case.,  One feature of the Connection Pool is to check if connections are still active based on an interval.  It is possible that the interval check could occur during the point the pool is in the process of shutting down.  This commit will change the behavior of throwing an exception during a liveliness check when the pool is shutting down, to now just ignore the failure to close cleanly.

The C++ exception informaton for SQLCode and SQLState is now going to be added to the C++ Exception message that is passed for the Node.js exception created on the other side of the NaN bridge.  The format will be JSON, so the new codes and the existing text are passed as properties that can be parsed in javascript.

For testing purposes, added an additional command to docker testing directives to allow environment variables to be passed from .env file, which can have variable=value format.

Pulled a couple SQLException tests out of the 14.errorhandling.js file, put them into a 16.SQLException.js file to make part of the smoke tests.

### CODE REVIEW

File: lib/pool.js [hunk 1]
```diff
@@ -244,10 +244,10 @@ class Pool {
   async _makeConnection() {
     this.counters.makeCall += 1;
     if (
-      Object.keys(this.all_connections).length > this.config.maxLimit ||
-      this.counters.closeCall != this.counters.closeExit
+      (process.env.PINO_LOG_ENABLED && (process.env.PINO_LOG_ENABLED == 'true' || process.env.PINO_LOG_ENABLED == '1')) && (Object.keys(this.all_connections).length > this.config.maxLimit ||
+      this.counters.closeCall != this.counters.closeExit)
     ) {
-      logger.level = "info";
+      logger.evel = "info";
```
Comment (adriansuarez) [inline lib/pool.js:250]: This looks like a typo.

File: .env
[the remark this answers is not in the document]
Comment (dwhite-nuodb) [reply .env:1]: I added it as something the docker commands to run tests, it is to add environment variables to the test runs, but at the moment tests will fail without the existence of the file.  I did not realize it would fail until CirleCi ran the tests without the file.  Perhaps I will address differently in the future, but this is a work around for now.

File: lib/pool.js (hunk 1 above)
Comment (dwhite-nuodb) [reply lib/pool.js]: You are right, slipped through since it is dynamic properties, so no compiler checks for this.  Made a commit just now to fix.  Thanks for noticing.

File: build-support/scripts/up [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-: ${IMG_NAME:="nuodb/nuodb:latest"}
+: ${IMG_NAME:="nuodb/nuodb:5.1"}
```
Comment (adriansuarez) [inline build-support/scripts/up:3]: @eresende-nuodb already made the latest release work. Can you revert this change and see if the CircleCI build passes?

Comment (dwhite-nuodb) [reply build-support/scripts/up:3]: I will change this back, since if probably make sense for overnights, I had left it that way since it kept failing for me.

**Review by prosenberg1 — APPROVED** [review]

**Review by adriansuarez — APPROVED** [review]
